### PR TITLE
Updating Instruments script, automation template, and framework for V2

### PR DESCRIPTION
**Pull request by jwallra**
Updating the instruments script and Automation template to remove app key, update devices available for test, and update WindowsAzureMobileServices.framework in the project to latest version (0.3.21).

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jwallra__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jawal). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: sdk/iOS/RunInstruments.sh [hunk 1]
```diff
@@ -7,41 +7,44 @@ mkdir Results
...
-if [ $8 ]
+if [ $7 ]
 then
   # Copy specified framework
-  cp -f $8 sdk.zip
+  cp -f $7 sdk.zip
 else
   # Copy in current version of the framework
-  curl --location --output sdk.zip --silent https://go.microsoft.com/fwLink/?LinkID=266533
+  curl --location --output sdk.zip https://zumo.blob.core.windows.net/sdk/azuresdk-iOS-v2.2.2.zip
```
Comment (phvannor) [inline sdk/iOS/RunInstruments.sh]: v2.2.2 is not compatible with MobileApps

File: sdk/iOS/RunInstruments.sh [hunk 2]
```diff
@@ -57,81 +60,104 @@ if [ "$DEVICE_CMD_ARG" == "iPad" ]; then
...
+  APP_NAME=$DIR/ZumoE2ETestApp/build/Release-iphonesimulator/ZumoE2ETestApp.app
+fi
+
+if [ "$DEVICE_CMD_ARG" == "iPhoneSim6s" ]; then
+  echo Using iPhone 6s Simulator
+  export DEVICE_ARG=iPhone\ 6s\ \(9.1\)
+  APP_NAME=$DIR/ZumoE2ETestApp/build/Release-iphonesimulator/ZumoE2ETestApp.app
+fi
+
+if [ "$DEVICE_CMD_ARG" == "iPhoneSim6sWatch" ]; then
```
Comment (phvannor) [inline sdk/iOS/RunInstruments.sh]: we dont have a watch test suite so not sure why we add it here, i guess its complete.

Comment (jwallra) [inline sdk/iOS/RunInstruments.sh]: I can remove it, seems harmless to have it for the future though...

File: sdk/iOS/RunInstruments.sh (hunk 1 above)
Comment (jwallra) [inline sdk/iOS/RunInstruments.sh]: This needs to be some variant of 3.0.0 then?

Comment (phvannor) [inline sdk/iOS/RunInstruments.sh]: Yup, 1 & 2 are services compatible, 3+ is for the Apps changes.

File: sdk/iOS/RunInstruments.sh (hunk 2 above)
Comment (phvannor) [inline sdk/iOS/RunInstruments.sh]: ideally, we need to work out iOS 8 testing to as well, but we can leave that to later.

File: sdk/iOS/RunInstruments.sh (hunk 1 above)
Comment (jwallra) [inline sdk/iOS/RunInstruments.sh]: Fixed, [see commit](https://github.com/jwallra/azure-mobile-services-test/commit/63d2ee21ddb25a9ddab4493832dd2b1d9ededeb9)

**Pull request merged**